File: programmers/level3/pro42886.py
# 프로그래머스 저울
# 제일 작은 애를 계속 더한 값 + 1 이 그 다음 값 보다 작으면 더한값 + 1 은 못 만드는 친구이다.
# 아니라면 다 더한 값 + 1 이 답
def solution(weight):
    weight.sort()
    temp = weight.pop(0)
    while(len(weight) != 0):
        v1 = weight.pop(0)
        if(temp + 1 < v1):
            return temp + 1 
        temp += v1
        
    return temp + 1

File: javascript/pro42886.js
// 프로그래머스 저울 

function solution(weight) {
    weight.sort((a,b) => a-b);
    let temp = 1;
    weight.forEach(value => {
        if (temp < value ) {
            return;
        }
        temp += value;
    })
    return temp;
}

let weight = [3,1,6,2,7,30,1];
console.log(solution(weight))